### DISCUSSION TITLE
Call to Undefined Function

### DISCUSSION THREAD

**Discussion by purchasingtoolpak** [flyntwp/flynt#427] [Q&A] [answered]
I am getting this error after deleting the Slider Image Gallery component. Both the dashboard get this error.

Fatal error: Uncaught Error: Call to undefined function Flynt\Components\SliderImageGallery\getACFLayout() in D:\Projects\Arc\LocalSites\app\public\wp-content\themes\flynt\inc\fieldGroups\postComponents.php on line 29

Line 190 is the first exception shown below:
`protected static function validateConfig($config, $requiredAttributes = [])
    {
        array_walk($requiredAttributes, function ($key) use ($config) {
            if (!array_key_exists($key, $config)) {
                throw new Exception("Field config needs to contain a \'{$key}\' property.");
            }
        });
        if (array_key_exists('key', $config)) {
            throw new Exception('Field config must not contain a \'key\' property.');
        }
        return $config;
    }`

Any help debugging would be appreciated.

**Comment by koukouchris** [comment, marked as answer]
Hello @purchasingtoolpak when deleting a component folder, as the error indicates in postComponents.php file remove `Components\SliderImages\getACFLayout(),` line.  
Search for all `Components\SliderImages\getACFLayout()` instances and delete those too.

**Reply by purchasingtoolpak** [reply]
I did delete the Components\SliderImageGallery\getACFLayout() if the post field group. The error message is in the ResolveConfig.php file. 

Uncaught Exception: Field config needs to contain a \'param\' property. in D:\Projects\Arc\LocalSites\app\public\wp-content\themes\flynt\vendor\flyntwp\acf-field-group-composer\lib\ACFComposer\ResolveConfig.php on line 190

Line 190 is the first exception shown below:
protected static function validateConfig($config, $requiredAttributes = []) { array_walk($requiredAttributes, function ($key) use ($config) { if (!array_key_exists($key, $config)) { throw new Exception("Field config needs to contain a \'{$key}\' property."); } }); if (array_key_exists('key', $config)) { throw new Exception('Field config must not contain a \'key\' property.'); } return $config; }

**Reply by koukouchris** [reply]
hi @purchasingtoolpak seems like 'param' is missing somewhere on a 'location' field where `registerFieldGroup` is used probably.

it should look like that:
```        
'location' => [
            [
                [
                      'param' => 'someParam', // is this missing or not correct?
                      'operator' => 'someOperator',
                      'value' => 'someValue'
                ],
            ]
        ]
```

**Reply by purchasingtoolpak** [reply]
Thanks so much for your help! The param was missing in the postComponents.php
